cmd/gossamer/configcmd.go

```diff
@@ -193,31 +194,46 @@ func loadStateAndRuntime(ss *state.StorageState, ks *keystore.Keystore) (*runtim
 	return runtime.NewRuntime(code, ss, ks)
 }
 
-// getConfig checks for config.toml if --config flag is specified and sets CLI flags
-func getConfig(ctx *cli.Context) (*dot.Config, error) {
-	currentConfig := gssmr.DefaultConfig()
+// getConfig gets the configuration for the node using --node and/or --config,
+// then applies the remaining cli flag options to the configuration
+func getConfig(ctx *cli.Context) (cfg *dot.Config, err error) {
+
+	// check --node flag and apply node defaults to config
+	if name := ctx.GlobalString(NodeFlag.Name); name != "" {
+		switch name {
+		case "gssmr":
+			log.Trace("[gossamer] Loading default configuration...", "node", name)
+			cfg = gssmr.DefaultConfig()
+		case "ksmcc":
+			log.Trace("[gossamer] Loading default configuration...", "node", name)
+			cfg = ksmcc.DefaultConfig()
+		}
+	} else {
+		log.Trace("[gossamer] Loading default configuration...", "node", "gssmr")
```

Comment:
maybe log that the node option isn't supported?

Reply:
gssmr is the default, this is not an error

Reply:
it's not an error but the user might have wanted to run a node that isn't supported, we should probably tell them it's not supported

Reply:
ah, I see, better to do that 3 lines up with `default` for the switch statement though